pkg/clients/handlers.go

```diff
@@ -29,12 +32,27 @@ type PaginationLinksResponse struct {
 	Next  string `json:"next,omitempty"`
 }
 
-func (a *API) RegisterEndpoints(mux *chi.Mux) {
+func (a *API) RegisterEndpoints(mux *chi.Mux) *API {
 	mux.Get("/api/v0/clients", a.ListClients)
 	mux.Post("/api/v0/clients", a.CreateClient)
 	mux.Get("/api/v0/clients/{id}", a.GetClient)
 	mux.Put("/api/v0/clients/{id}", a.UpdateClient)
 	mux.Delete("/api/v0/clients/{id}", a.DeleteClient)
+
+	return a
+}
+
+func (a *API) SetupValidation(v *validation.Validator) *API {
+	err := v.RegisterValidatingFunc("clients", a.validatingFunc)
+	if err != nil {
+		a.logger.Fatal("unexpected validatingFunc already registered for clients")
+	}
+
+	return a
+}
+
+func (a *API) validatingFunc(r *http.Request) validator.ValidationErrors {
```

Comment:
Just a non-blocking question. Is this where the `a.validator` will do the validation logic?

Reply:
Yes, this is where the API struct will perform its validations leveraging the functionality from the validator object (3rd party)